### ISSUE TITLE
Design Patterns to work with with single pattern transforms

### ISSUE THREAD

**Issue by kjellr** [WordPress/gutenberg#30631]
Related to https://github.com/WordPress/gutenberg/pull/30469, https://github.com/WordPress/gutenberg/issues/27575, and #28736.

Single blocks will soon have the ability to transform into block patterns. To help introduce this feature we should design some new patterns that use just a single block. 

<img width="683" alt="Screen Shot 2021-04-08 at 1 51 34 PM" src="https://user-images.githubusercontent.com/1202812/114073851-c17f3d80-9871-11eb-8a79-1fdd3dc82b26.png">

Let's use this thread to brainstorm which individual blocks could benefit from this feature the most, and then to iterate on some pattern designs. 

cc @melchoyce

**Comment by melchoyce** [comment]
I've been trying to work through some ideas, but I keep running into a mental barrier where everything I can think of is just... a block style.

For example, two variations I can think of for list blocks? Setting `ul` or `ol`. For quotes, I'd offer the default styles. For tables, row stripes or plain, or maybe one of the colors pre-selected. These don't feel any different from what we offer now as block styles, and I'm struggling to break beyond that. Anything more feels too opinionated on a design level, since most of these are design customizations, not just layout.

**Comment by kjellr** [comment]
Yeah, I definitely understand the potential confusion between Block Patterns and Block Styles. Also, just the other day [I ran into a scenario](https://github.com/WordPress/gutenberg/pull/30530#issuecomment-816792315) where I think converting one of the current bundled Block Styles into a pattern would make more sense. 

@jasmussen or @jameskoster do either of you have any philosophical thoughts on the difference between Styles and Patterns here? I'm wondering if (where possible) we should begin translating some of our existing block styles into Block Patterns instead.

**Comment by jameskoster** [comment]
I may be totally off-base, but I would anticipate that as more design tools (https://github.com/WordPress/gutenberg/issues/27331) come online, Patterns will largely replace Styles in terms of how they work under the hood. 

This has the obvious benefit of enabling people to effectively create their own theme-agnostic block styles directly in the editor, then share them via the pattern directory, across sites, and so on. 

That's not to say that single-block patterns like these couldn't still be _presented_ using the block styles UI of today. That part feels less clear.

**Comment by jasmussen** [comment]
I would echo Jay that the key here is that looking at what design tools we have available, and surfacing curated combinations of those, is our goal here. As usable as we can make the inspector, ultimately figuring out what combination of allcaps, line-height, letter-spacing, margin, font-weight looks good, is where these patterns can come in.

Perhaps a helpful way of thinking about paragraph patterns is to think of Figma font presets. Simply toggling "all-caps" is likely not going to inspire anyone. But if a style can be surfaced visually that uses all-caps, sets the font size smaller, the letter spacing a bit wider, expands the top margin a bit, suddenly you might have a good looking sub-heading style. Which, even if it's just a shortcut to existing tools, it's a _curated shortcut_.

Again to Jay's point, we have an Image style called _rounded_, which really just sets the border radius to 9999px (because that makes landscape or portrait images pill-shaped rather than ellipsoid). Potentially I imagine this style likely being deprecated entirely in favor of a pattern that does the same. 

The balance to strike between pattern and style is to think about performance, markup, and _doing things right_. If our design tools can avoid inline styles, patterns that leverage them will create good and performant markup. As soon as inline styles get involved, or the complexity of the treatment becomes sufficient, a _style_ is likely the better choice.
